ext/dom/element.c

```diff
@@ -1715,6 +1715,98 @@ PHP_METHOD(Dom_Element, insertAdjacentText)
 }
 /* }}} end DOMElement::insertAdjacentText */
 
+/* https://html.spec.whatwg.org/#dom-element-insertadjacenthtml */
+PHP_METHOD(Dom_Element, insertAdjacentHTML)
+{
+	zval *where_zv;
+	zend_string *string;
+
+	dom_object *this_intern;
+	zval *id;
+	xmlNodePtr thisp;
+
+	bool created_context = false;
+
+	ZEND_PARSE_PARAMETERS_START(2, 2)
+		Z_PARAM_OBJECT_OF_CLASS(where_zv, dom_adjacent_position_class_entry)
+		Z_PARAM_STR(string)
+	ZEND_PARSE_PARAMETERS_END();
+
+	DOM_GET_THIS_OBJ(thisp, id, xmlNodePtr, this_intern);
+
+	const zend_string *where = Z_STR_P(zend_enum_fetch_case_name(Z_OBJ_P(where_zv)));
+
+	/* 1. We don't do injection sinks. */
+
+	/* 2. Let context be NULL */
+	xmlNodePtr context = NULL;
+
+	/* 3. Use the first matching item from this list: (...) */
+	switch (ZSTR_LEN(where) + ZSTR_VAL(where)[2]) {
+		case sizeof("beforebegin") - 1 + 'f':
```

Comment:
```suggestion
		case sizeof("BeforeBegin") - 1 + 'f':
```
Same elsewhere. It is confusing to see the lowercased name, even if it is used only for the length.

Reply:
That's fair, I'll change it.